src/service/repo/anonymousMessageRepo.ts

```diff
@@ -150,11 +154,16 @@ async function setChannelAsRead(channelId: string): Promise<boolean> {
   }
 }
 
-async function getAnonymousChannelDetail(channelType: GetstreamChannelType, channelId: string) {
+async function getAnonymousChannelDetail(
+  channelType: GetstreamChannelType,
+  channelId: string,
+  timeStamp?: string
+) {
+  const url = timeStamp
+    ? `${baseUrl.getAnonymousChannelDetail(channelType, channelId)}?last_fetch_date=${timeStamp}`
+    : baseUrl.getAnonymousChannelDetail(channelType, channelId);
   try {
-    const response = await anonymousApi.get(
-      baseUrl.getAnonymousChannelDetail(channelType, channelId)
-    );
+    const response = await anonymousApi.get(url);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The conditional construction of the URL based on the presence of `timeStamp` is handled well. However, there seems to be no error handling if the `anonymousApi.get(url)` call fails. Consider adding a catch block to handle potential errors from this API call.
```diff
   try {
    const response = await anonymousApi.get(url);
    if (response.status === 200) {
      return Promise.resolve(response.data?.data);
    }
+    } catch (error) {
+      console.error('Error fetching anonymous channel detail:', error);
+      return Promise.reject(error);
+    }
```
This will ensure that any issues with the API call are logged and properly communicated back to the caller.


<!-- This is an auto-generated comment by OSS CodeRabbit -->